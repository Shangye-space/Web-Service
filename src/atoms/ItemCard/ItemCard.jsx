import React from 'react';
import { Link } from 'react-router-dom';
import classnames from 'classnames';

import styles from './ItemCard.module.scss';

const ItemCard = ({ name, price, id }) => {
    return (
        <div className={classnames('col-3', 'hvr-grow', styles.wrapper)}>
            <Link to={`/shop/item/${id}`} className={styles.link}>
                <div className={styles.card}>
                    <img
                        src="https://s13.postimg.cc/5al19x0d3/media_img.jpg"
                        className={styles.cardMedia}
                    />
                    <div className={styles.cardDetails}>
                        <h2 className={styles.cardHead}>{name}</h2>

                        <strong className={styles.cardActionButton}>
                            {price}$
                        </strong>
                    </div>
                </div>
            </Link>
        </div>
    );
};

export default ItemCard;
