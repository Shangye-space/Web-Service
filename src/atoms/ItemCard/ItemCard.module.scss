@import '../../settings/variables.scss';

.wrapper {
    padding: 1em 0;
    min-width: 277.5px;

    & :hover {
        cursor: pointer;
    }
}

.link {
    text-decoration: none !important;
    color: $default-dark !important;

    & :hover {
        text-decoration: none !important;
        color: $default-dark !important;
    }
}

.clearfix:before,
.clearfix:after {
    content: ' ';
    display: table;
}

.clearfix:after {
    clear: both;
}

.card {
    background-color: #fff;
    border-radius: 2px;
    box-shadow: -1px 2px 4px rgba(0, 0, 0, 0.25), 1px 1px 3px rgba(0, 0, 0, 0.1);
    margin: 0 auto;
    max-width: 340px;
}

.cardMedia {
    width: 100%;
}

.cardDetails {
    padding: 5px 15px;
}

.cardHead {
    font-weight: 400;
    font-size: 26px;
    margin-bottom: 15px;
}

.cardActionButton {
    border-radius: 2px;
    color: $default-dark;
    cursor: pointer;
    display: inline-block;
    font-weight: 500;
    margin: 0;
    padding: 10px;
    text-decoration: none;

    & :hover {
        background-color: rgba(255, 168, 58, 0.2);
    }
}
