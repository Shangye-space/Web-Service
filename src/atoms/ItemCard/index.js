import ItemCard from './ItemCard.jsx';

export default ItemCard;
