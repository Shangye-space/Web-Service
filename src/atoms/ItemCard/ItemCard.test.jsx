import React from 'react';
import ReactDOM from 'react-dom';
import { BrowserRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';

import ItemCard from './ItemCard.jsx';

it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
        <Router>
            <ItemCard />
        </Router>,
        div,
    );
    ReactDOM.unmountComponentAtNode(div);
});

it('snapshot renders', () => {
    const component = renderer.create(
        <Router>
            <ItemCard />
        </Router>,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});
