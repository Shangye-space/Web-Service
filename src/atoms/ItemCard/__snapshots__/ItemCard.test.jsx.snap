// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot renders 1`] = `
<div
  className="col-3 hvr-grow wrapper"
>
  <a
    className="link"
    href="/shop/item/undefined"
    onClick={[Function]}
  >
    <div
      className="card"
    >
      <img
        className="cardMedia"
        src="https://s13.postimg.cc/5al19x0d3/media_img.jpg"
      />
      <div
        className="cardDetails"
      >
        <h2
          className="cardHead"
        />
        <strong
          className="cardActionButton"
        >
          $
        </strong>
      </div>
    </div>
  </a>
</div>
`;
